scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ForkedTaskVariablesManager.java

```diff
@@ -0,0 +1,157 @@
+import java.util.Map;
+
+import org.ow2.proactive.scheduler.common.task.TaskResult;
+import org.ow2.proactive.scheduler.common.task.util.SerializationUtil;
+import org.ow2.proactive.scheduler.common.util.VariableSubstitutor;
+import org.ow2.proactive.scheduler.task.context.TaskContext;
+import org.ow2.proactive.scripting.Script;
+import org.ow2.proactive.scripting.ScriptHandler;
+import org.ow2.proactive.scripting.TaskScript;
+
+public class ForkedTaskVariablesManager implements Serializable {
+    public static final String MULTI_NODE_TASK_NODESURL_BINDING_NAME = "nodesurl";
+    public static final String VARIABLES_BINDING_NAME = "variables";
+    public static final String DS_SCRATCH_BINDING_NAME = "localspace";
+    public static final String DS_INPUT_BINDING_NAME = "inputspace";
+    public static final String DS_OUTPUT_BINDING_NAME = "outputspace";
+    public static final String DS_GLOBAL_BINDING_NAME = "globalspace";
+    public static final String DS_USER_BINDING_NAME = "userspace";
+    public static final String FORK_ENVIRONMENT_BINDING_NAME = "forkEnvironment";
+    /**
+     * Will be replaced by the matching third-party credential
+     * Example: if one of the third-party credentials' key-value pairs is 'foo:bar',
+     * then '$credentials_foo' will be replaced by 'bar' in the arguments of the tasks.
+     */
+    public static final String CREDENTIALS_KEY_PREFIX = "credentials_";
+
+    private static TaskResult[] tasksResults(TaskContext container) {
+        TaskResult[] previousTasksResults = container.getPreviousTasksResults();
+        if (previousTasksResults != null) {
+            return previousTasksResults;
+        } else {
+            return new TaskResult[0];
+        }
+    }
+
+
+    public void addBindingsToScriptHandler(ScriptHandler scriptHandler, TaskContext taskContext,
+            Map<String, Serializable> variables, Map<String, String> thirdPartyCredentials) {
+        scriptHandler.addBinding(VARIABLES_BINDING_NAME, variables);
+
+        scriptHandler.addBinding(TaskScript.RESULTS_VARIABLE, tasksResults(taskContext));
+
+        scriptHandler.addBinding(TaskScript.CREDENTIALS_VARIABLE, thirdPartyCredentials);
+
+        scriptHandler.addBinding(DS_SCRATCH_BINDING_NAME, taskContext.getScratchURI());
+        scriptHandler.addBinding(DS_INPUT_BINDING_NAME, taskContext.getInputURI());
+        scriptHandler.addBinding(DS_OUTPUT_BINDING_NAME, taskContext.getOutputURI());
+        scriptHandler.addBinding(DS_GLOBAL_BINDING_NAME, taskContext.getGlobalURI());
+        scriptHandler.addBinding(DS_USER_BINDING_NAME, taskContext.getUserURI());
+
+        scriptHandler.addBinding(MULTI_NODE_TASK_NODESURL_BINDING_NAME, taskContext.getOtherNodesURLs());
+
+        scriptHandler.addBinding(FORK_ENVIRONMENT_BINDING_NAME,
+                taskContext.getInitializer().getForkEnvironment());
+    }
+
+    public Map<String, String> extractThirdPartyCredentials(TaskContext container) throws Exception {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)